### PR TITLE
nec/pce.cpp, nec/pce_cd.cpp: Various updates

### PR DESCRIPTION
**Pull request by cam900**
- Use device_mixer_interface rather than using hard-coded speaker tag
- Reduce defines
- Remove outdated comments
- Use lowercase hexadecimal values
- Remove nec/pce_m.cpp, Move pce_m.cpp codes into nec/pce.cpp

### CODE REVIEW

File: src/mame/nec/pce.cpp [hunk 1]
```diff
@@ -69,27 +69,62 @@ Super System Card:
...
+	m_port_ctrl->sel_w(BIT(data, 0));
+	m_port_ctrl->clr_w(BIT(data, 1));
+}
+
+u8 pce_state::controller_r()
+{
+	u8 ret = (m_port_ctrl->port_r() & 0x0f) | m_io_port_options;
+#ifdef UNIFIED_PCE
+	ret &= ~0x40;
+#endif
```
Comment (angelosa) [inline src/mame/nec/pce.cpp:96]: Looks like this `UNIFIED_PCE` macro is unused, it's safe to remove.

**Comment by cuavas** [comment]
What’s the motivation for moving it into src/devices/machine?  Is it used for anything besides the PC-Engine?

**Comment by angelosa** [comment]
File is currently a mixup of _PC8801_-30 drive, the specific (and opaque) PC Engine SCSI controller and the rest of the expansion unit bus & i/f.

https://github.com/mamedev/mame/issues/11078

afaik _PC8801_-31 SCSI isn't much different beyond relying on SEL line alone (and doing the rest internally), a preliminary core i/f (without the controller part, _maybe_ at ID 7):

https://github.com/mamedev/mame/blob/91037294f03b4abb5a38eb161f1ee7a1438d68eb/src/devices/bus/pc8801/pc8801_31.cpp

Bottom line: not convinced about the move either, it's essentially a NEC i/f that can stay in `src/mame/nec`.

**Review by angelosa — APPROVED** [review]
